### ISSUE TITLE
[Feature request] Support adaptive softmax

### ISSUE THREAD

**Issue by BrikerMan** [BrikerMan/Kashgari#114]
Adaptive softmax could provide up to 10x speed performance. And here is the keras implements [CyberZHG/keras-adaptive-softmax](https://github.com/CyberZHG/keras-adaptive-softmax).

**Additional context**

* [Efficient softmax approximation for GPUs](https://arxiv.org/abs/1609.04309)
